src/worker.ts

```diff
@@ -0,0 +1,28 @@
+import BN from 'bn.js'
+import workerpool from 'workerpool'
+import { OrderbookFetcher } from './orderbook_fetcher'
+import { sortOrderbookBySerializedPrice } from './utilities'
+import { transitiveOrderbook } from '@gnosis.pm/dex-contracts'
+
+function markets(orderbooks_object: object, base: string, quote: string, hops: number) {
+  const orderbooks = OrderbookFetcher.deserializeOrderbooks(orderbooks_object);
+  const transitive = transitiveOrderbook(orderbooks, base, quote, hops);
+  const result = sortOrderbookBySerializedPrice(transitive);
+  return result;
+}
+
+function estimatedBuyAmount(orderbooks_object: object, base: string, quote: string, hops: number, quoteAmount: string) {
+  const orderbooks = OrderbookFetcher.deserializeOrderbooks(orderbooks_object);
+  const transitive = transitiveOrderbook(orderbooks, base, quote, hops);
+  const sellAmount = new BN(quoteAmount);
+  // The orderbook API only allows us to compute a price for selling base tokens. Here, we want to sell quote tokens.
+  // Thus, we invert the orderbook, sell the requested amount (now in base) and multiply the computed price with the
+  // sell amount in order to compute the buy amount in the original base token. A small rounding buffer is considered
+  // to increase the chances of matching the order if placed.
```

Comment:
Can we either apply the rounding buffer here (rather than in the handler) or move all price estimation logic back into the handler. The only expensive thing should be computing the transitive orderbbok since this has quadratic runtime.

Reply:
Then the worker pool is only used for computing the transitive orderbook? That sounds OK to me.